packages/pigeon/example/README.md

```diff
@@ -1,155 +1,277 @@
+<?code-excerpt path-base="excerpts/packages/pigeon_example"?>
 # Pigeon Examples
 
+The examples here will cover basic usage. For a more thorough set of examples,
+check the [core_tests pigeon file](../pigeons/core_tests.dart) and 
+[platform test folder](../platform_tests/) ([shared_test_plugin_code](../platform_tests/shared_test_plugin_code/) and [alternate_language_test_plugin](../platform_tests/alternate_language_test_plugin/) especially).
+
+## Invocation
+
+Begin by configuring pigeon at the top of the `.dart` input file.
+In actual use, you would include only the languages
+needed for your project.
+
+<?code-excerpt "../../app/pigeons/messages.dart (config)"?>
+```dart
+@ConfigurePigeon(PigeonOptions(
+  dartOut: 'lib/src/messages.g.dart',
+  dartOptions: DartOptions(),
+  cppOptions: CppOptions(namespace: 'pigeon_example'),
+  cppHeaderOut: 'windows/runner/messages.g.h',
+  cppSourceOut: 'windows/runner/messages.g.cpp',
+  kotlinOut:
+      'android/app/src/main/kotlin/dev/flutter/pigeon_example_app/Messages.g.kt',
+  kotlinOptions: KotlinOptions(),
+  javaOut: 'android/app/src/main/java/io/flutter/plugins/Messages.java',
+  javaOptions: JavaOptions(),
+  swiftOut: 'ios/Runner/Messages.g.swift',
+  swiftOptions: SwiftOptions(),
+  objcHeaderOut: 'macos/Runner/messages.h',
+  objcSourceOut: 'macos/Runner/messages.m',
+  objcOptions: ObjcOptions(prefix: 'PGN'),
+  copyrightHeader: 'pigeons/copyright.txt',
+))
+```
+Then make a simple call to run pigeon on the Dart file containing your definitions.
+
+```sh
+flutter pub run pigeon --input path/to/input.dart
+```
+
 ## HostApi Example
 
 This example gives an overview of how to use Pigeon to call into the
-host-platform from Flutter.
-
-For instructions to set up your own Pigeon usage see these [steps](https://pub.dev/packages/pigeon#usage).
+host platform from Flutter.
 
-### message.dart
+### Dart input
 
 This is the Pigeon file that describes the interface that will be used to call
 from Flutter to the host-platform.
 
+<?code-excerpt "../../app/pigeons/messages.dart (host-definitions)"?>
 ```dart
-import 'package:pigeon/pigeon.dart';
-
-class Book {
-  String? title;
-  String? author;
+enum Code { one, two }
+
+class MessageData {
+  MessageData({required this.code, required this.data});
+  String? name;
+  String? description;
+  Code code;
+  Map<String?, String?> data;
 }
 
 @HostApi()
-abstract class BookApi {
-  List<Book?> search(String keyword);
+abstract class ExampleHostApi {
+  String getHostLanguage();
+
+  @ObjCSelector('addNumber:otherNumber:')
+  @SwiftFunction('add(number:otherNumber:)')
+  int add(int a, int b);
+
+  @async
+  bool sendMessage(MessageData message);
 }
 ```
 
```

Comment:
I think a sentence or two explaining `ObjCSelector`/`SwiftFunction` would help here, since there's no context provided for what they are or why they are there currently. Or alternately, a comment above those lines in the code explaining what they do.

Reply:
I mentioned them (without much information) in the section about defining your definition files in the main README